### PR TITLE
[22769] Clickable and Text Copy Appt Card

### PR DESCRIPTION
**Pull request by LanceGundersen**
## Description
Implement VSP recommendation (in [#18583](https://github.com/department-of-veterans-affairs/va.gov-team/issues/18583)) - full cards must be accessible to screen readers and allow users to copy text.

## Testing done
Local

## Acceptance criteria
- [ ] Changes are behind va_online_scheduling_homepage_refresh feature flag
- [ ] If user clicks/taps anywhere on the appointment card, Then the appointment details open
- [ ] User can copy text
- [ ] Fix is reviewed by VSP Accessibility

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [ ] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: src/applications/vaos/appointment-list/components/AppointmentsPageV2/AppointmentListItem.jsx [hunk 1]
```diff
@@ -69,85 +69,98 @@ function VAFacilityName({ facility }) {
...
   const canceled = appointment.status === APPOINTMENT_STATUS.cancelled;
+  const link = isCommunityCare
+    ? `cc/${appointment.id}`
+    : `va/${appointment.id}`;
 
   return (
     <li
       data-request-id={appointment.id}
-      className="vaos-appts__card vads-u-display--flex vads-u-align-items--center"
+      className="vaos-appts__card vaos-appts__card--clickable"
```
Comment (jbalboni) [inline src/applications/vaos/appointment-list/components/AppointmentsPageV2/AppointmentListItem.jsx:87]: You might consider removing the padding from this element and adding it to the div below. The way it is now, the whole card isn't clickable, just the inner area immediately surrounding the text.

Green area isn't clickable:
Screen Shot 2021-04-08 at 2 03 16 PM

**Review by jbalboni — COMMENTED** [review]
One comment about the clickable area, but looks good otherwise

**Comment by LanceGundersen** [comment]
> One comment about the clickable area, but looks good otherwise

@jbalboni Good catch, I messed with the styling and didn't see that edge case. Fixed now.

**Review by 1Copenut — CHANGES_REQUESTED** [review]
I'm stoked about this change @LanceGundersen. I tested with Safari + VO and have just one change request. The list items are announcing as "group" because of how VO treats list items that have had the styles updated. I'd like to add a `role="list"` to the UL like so:

```diff
<ul
+ role="list"
  aria-labelledby={`appointment_list_${monthDate.format('YYYY-MM',)}`}
  className="vads-u-padding-left--0"
>
```

ESLint is going to complain about it, but we can add a comment to disable the rule. It's redundant but causes VO to announce the cards as list items which is what we want.

**Comment by LanceGundersen** [comment]
@jbalboni can you verify the clickable area is fixed?

File: src/applications/vaos/appointment-list/sass/styles.scss [hunk 2]
```diff
@@ -69,6 +69,24 @@
     padding: 24px;
     margin-bottom: 24px;
   }
+
+  &.vaos-appts__card--clickable {
+    list-style: none;
+    padding: 0;
+
+    .vaos-appts__card--clickable-content {
```
Comment (jbalboni) [inline src/applications/vaos/appointment-list/sass/styles.scss:77]: You could make the functionality in this class with the design system classes: `vads-u-padding--2 medium-screen:vads-u-padding--3 medium-screen:vads-u-margin-bottom--3`

**Review by jbalboni — APPROVED** [review]
LGTM. I made one comment about styles, but I don't feel that strongly about it, you don't have to make those changes to merge this PR.

File: src/applications/vaos/appointment-list/sass/styles.scss (hunk 2 above)
Comment (LanceGundersen) [reply src/applications/vaos/appointment-list/sass/styles.scss]: @jbalboni makes sense not to add the bloat. I am leveraging the design system classes now. Just take a peek again to verify nothing got asked on us.

**Review by jbalboni — APPROVED** [review]

**Review by 1Copenut — APPROVED** [review]
👍 LGTM! Let me know when this is in staging and I'll test in on NVDA and JAWS.

**Review by jhonny218 — APPROVED** [review]
LGTM

**Pull request merged**